sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/quickpulse/filtering/DerivedMetricProjections.java

```diff
@@ -78,17 +67,8 @@ public void calculateProjection(DerivedMetricInfo derivedMetricInfo, TelemetryCo
     }
 
     private void calculateAggregation(AggregationType type, String id, double incrementBy) {
-        synchronized (lock) {
-            DerivedMetricAggregation dma = derivedMetricValues.get(id);
-            dma.count++;
-            if (type.equals(AggregationType.SUM) || type.equals(AggregationType.AVG)) {
-                dma.aggregation += incrementBy;
-            } else if (type.equals(AggregationType.MIN)) {
-                dma.aggregation = Math.min(dma.aggregation, incrementBy);
-            } else if (type.equals(AggregationType.MAX)) {
-                dma.aggregation = Math.max(dma.aggregation, incrementBy);
-            }
-        }
+        DerivedMetricAggregation dma = derivedMetricValues.get(id);
+        dma.update(incrementBy);
     }
 
     private static class DerivedMetricAggregation {
```

Comment:
one more thing I think is to move `DerivedMetricAggregation` to its own file and make the fields private, so it's clear they can't be accessed outside of the properly synchronized accessors